Firestore/Source/Core/FSTSyncEngine.mm

```diff
@@ -114,6 +114,26 @@ - (instancetype)initWithQuery:(FSTQuery *)query
 
 @end
 
+#pragma mark - LimboResolution
+
+/** Tracks a limbo resolution. */
+class LimboResolution {
+ public:
+  LimboResolution() : key{DocumentKey()} {
```

Comment:
`DocumentKey` as a non-POD type will default initialize itself so you don't need to explicitly default initialize `key`. `bool` is a POD type so it has no default initializer, so you *do* need to initialize it.

Rather than repeat the initialization of `documentReceived`, add a default assignment below (`bool document_received = false;`).

With that in place, this can just be
```c++
LimboResolution() {
}
```